projects/core/src/error-handling/http-error-handler/http-error-handler.interceptor.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ErrorHandler, Injectable } from '@angular/core';
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Injectable()
```

Comment:
Let's add a bit of JSDocs. Just for inspiration:
```
This interceptor forwards all HTTP errors (e.g. 5xx or 4xx status response from backend)
to Angular's `ErrorHandler`.

Thanks to this, in SSR, any HTTP error from backend can potentially mark the Server-Side Rendering
as errord and therefore allow for sending an appropriate error response to a final client of SSR.
```

Reply:
thanks, I've added jsdocs